File: app/src/main/java/com/robotnec/budget/core/calculator/CalculatorModel.java
package com.robotnec.budget.core.calculator;

import com.robotnec.budget.core.exception.InvalidExpressionException;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface CalculatorModel {
    String digit(int digit);

    String dot();

    String calculate() throws InvalidExpressionException;

    String operation(Op op);

    String back();

    String clear();
}

File: app/src/main/java/com/robotnec/budget/core/calculator/CalculatorModel.kt
package com.robotnec.budget.core.calculator

import com.robotnec.budget.core.exception.InvalidExpressionException

/**
 * @author zak zak@swingpulse.com>
 */
interface CalculatorModel {
    fun digit(digit: Int): String

    fun dot(): String

    @Throws(InvalidExpressionException::class)
    fun calculate(): String

    fun operation(op: Op): String

    fun back(): String

    fun clear(): String
}
